src/main/java/com/xpanxion/java/springboot/da1/demo/service/student5/WorkoutHistoryService5.java

```diff
@@ -32,4 +37,32 @@ public WorkoutHistory5 workoutCheckOut(Long memberId, Date time) {
         return workoutHistoryRepository.findAllByMemberMemberId(member.getMemberId()).stream()
                 .map(c -> new WorkoutHistoryView5(c.getMember().getMemberId(), c.getTimeUtc(), c.getCheckType())).toList();
     }
+
+    public WorkoutLengthView findMinMaxByMemberId(Long memberId, String type) {
+        List<WorkoutLengthView> workoutLengths = new ArrayList<>();
+        memberService.validateMember(memberId);
+        Format formatter = new SimpleDateFormat("MM/dd/yy");
+        var workoutHistory = workoutHistoryRepository.findAllByMemberMemberId(memberId);
+        for (int i = 0; i < workoutHistory.size(); i+=2) {
+            String date = formatter.format(workoutHistory.get(i).getTimeUtc());
+            var lengthInMinutes = getDateDiff(workoutHistory.get(i).getTimeUtc(), workoutHistory.get(i+1).getTimeUtc());
+            workoutLengths.add(new WorkoutLengthView(lengthInMinutes, memberId, date));
+        }
+        switch (type) {
+            case "min" -> {
+                return workoutLengths.get(0);
+            }
+            case "max" -> {
+                Collections.sort(workoutLengths);
+                return workoutLengths.get(0);
+            }
+            default -> throw new ResponseStatusException(HttpStatus.NOT_FOUND, "type is invalid");
+        }
+    }
+
+    //Helper Methods
+    public static long getDateDiff(Date date1, Date date2) {
```

Comment:
It's probably best to make this private.

Reply:
Is it ideal to solve all issues in one push or break them up like I just did.